src/app/server/Server.cpp

```diff
@@ -651,6 +651,9 @@ void Server::Shutdown()
 #if CHIP_CONFIG_ENABLE_ICD_SERVER
     app::DnssdServer::Instance().SetICDManager(nullptr);
 #endif // CHIP_CONFIG_ENABLE_ICD_SERVER
+
+    app::InteractionModelEngine::GetInstance()->Shutdown();
```

Comment:
I would expect shutdown to just happen in reverse order from init, though this code seems to be doing a terrible job of that.

But just in terms of safety, why are we moving where IM engine shutdown happens?  Why didn't we just add the new call sometime after that point?

Reply:
Revered the shutdown sequence and use an existing API to `ShutdownActiveReads` which was guarded by CONFIG_BUILD_FOR_HOST_UNIT_TEST